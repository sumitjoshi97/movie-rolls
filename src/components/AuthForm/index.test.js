import React from 'react'
import { shallow } from 'enzyme'

import { AuthForm } from './index'

describe('AuthForm unit test', () => {
  const signupUser = jest.fn()
  const loginUserWithEmail = jest.fn()
  const loginUserWithSocial = jest.fn()

  const props = {
    signupUser,
    loginUserWithEmail,
    loginUserWithSocial
  }

  const authFormWrapper = shallow(<AuthForm {...props} />)

  test('should render properly', () => {
    expect(authFormWrapper).toMatchSnapshot()
  })

  test('clicking signup button changes auth to signup', () => {
    authFormWrapper.find('.auth-form__active').simulate('click')
    expect(authFormWrapper.state().active).toEqual('signup')
  })

  describe('submitting form will send user info', () => {
    const email = 'test@test.com'
    const password = 'test123'

    test('should return email', () => {
      authFormWrapper.find('input[name="email"]').simulate('change', {
        target: { name: 'email', value: email }
      })
      expect(authFormWrapper.state().email).toEqual(email)
    })

    test('should return password', () => {
      authFormWrapper.find('input[name="password"]').simulate('change', {
        target: { name: 'password', value: password }
      })
      expect(authFormWrapper.state().password).toEqual(password)
    })

    test('should call signupUser fn with email and password', () => {
      authFormWrapper.find('.auth-form__form__submit').simulate('click', {
        preventDefault: () => {}
      })
      expect(signupUser).toHaveBeenCalledWith(email, password)
    })

    test('should change form to login on click signup', () => {
      authFormWrapper.find('.auth-form__active').simulate('click')
      expect(authFormWrapper.state().active).toEqual('login')
    })

    test('should call login user fn. with email and password', () => {
      authFormWrapper.find('.auth-form__form__submit').simulate('click', {
        preventDefault: () => {}
      })
      expect(signupUser).toHaveBeenCalledWith(email, password)
    })
  })
})
