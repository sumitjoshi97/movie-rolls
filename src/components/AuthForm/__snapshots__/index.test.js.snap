// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm unit test should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthForm
    loginUserWithEmail={[MockFunction]}
    loginUserWithSocial={[MockFunction]}
    signupUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="auth-form__header"
        >
          <h3>
            Log in
             to see more
          </h3>
        </div>,
        <form
          className="auth-form__form"
        >
          <input
            className="auth-form__form__input email"
            name="email"
            onChange={[Function]}
            placeholder="Email address"
            type="email"
            value=""
          />
          <input
            className="auth-form__form__input password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <input
            className="auth-form__form__submit action-btn"
            onClick={[Function]}
            type="submit"
            value="Log in"
          />
        </form>,
        <span>
          Or
        </span>,
        <div
          className="auth-form__social-btns"
        >
          <button
            className="auth-form__social-btns__google action-btn"
            onClick={[MockFunction]}
          >
            Continue with Google
          </button>
        </div>,
        <button
          className="auth-form__active"
          onClick={[Function]}
        >
          Not on Movie Rolls? Sign up
        </button>,
      ],
      "className": "auth-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3>
            Log in
             to see more
          </h3>,
          "className": "auth-form__header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Log in",
              " to see more",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Log in",
            " to see more",
          ],
          "type": "h3",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="auth-form__form__input email"
              name="email"
              onChange={[Function]}
              placeholder="Email address"
              type="email"
              value=""
            />,
            <input
              className="auth-form__form__input password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />,
            <input
              className="auth-form__form__submit action-btn"
              onClick={[Function]}
              type="submit"
              value="Log in"
            />,
          ],
          "className": "auth-form__form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "auth-form__form__input email",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email address",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "auth-form__form__input password",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "auth-form__form__submit action-btn",
              "onClick": [Function],
              "type": "submit",
              "value": "Log in",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Or",
        },
        "ref": null,
        "rendered": "Or",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="auth-form__social-btns__google action-btn"
            onClick={[MockFunction]}
          >
            Continue with Google
          </button>,
          "className": "auth-form__social-btns",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Continue with Google",
            "className": "auth-form__social-btns__google action-btn",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Continue with Google",
          "type": "button",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Not on Movie Rolls? Sign up",
          "className": "auth-form__active",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Not on Movie Rolls? Sign up",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="auth-form__header"
          >
            <h3>
              Log in
               to see more
            </h3>
          </div>,
          <form
            className="auth-form__form"
          >
            <input
              className="auth-form__form__input email"
              name="email"
              onChange={[Function]}
              placeholder="Email address"
              type="email"
              value=""
            />
            <input
              className="auth-form__form__input password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <input
              className="auth-form__form__submit action-btn"
              onClick={[Function]}
              type="submit"
              value="Log in"
            />
          </form>,
          <span>
            Or
          </span>,
          <div
            className="auth-form__social-btns"
          >
            <button
              className="auth-form__social-btns__google action-btn"
              onClick={[MockFunction]}
            >
              Continue with Google
            </button>
          </div>,
          <button
            className="auth-form__active"
            onClick={[Function]}
          >
            Not on Movie Rolls? Sign up
          </button>,
        ],
        "className": "auth-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              Log in
               to see more
            </h3>,
            "className": "auth-form__header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Log in",
                " to see more",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Log in",
              " to see more",
            ],
            "type": "h3",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="auth-form__form__input email"
                name="email"
                onChange={[Function]}
                placeholder="Email address"
                type="email"
                value=""
              />,
              <input
                className="auth-form__form__input password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />,
              <input
                className="auth-form__form__submit action-btn"
                onClick={[Function]}
                type="submit"
                value="Log in"
              />,
            ],
            "className": "auth-form__form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "auth-form__form__input email",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email address",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "auth-form__form__input password",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "auth-form__form__submit action-btn",
                "onClick": [Function],
                "type": "submit",
                "value": "Log in",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Or",
          },
          "ref": null,
          "rendered": "Or",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="auth-form__social-btns__google action-btn"
              onClick={[MockFunction]}
            >
              Continue with Google
            </button>,
            "className": "auth-form__social-btns",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Continue with Google",
              "className": "auth-form__social-btns__google action-btn",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Continue with Google",
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Not on Movie Rolls? Sign up",
            "className": "auth-form__active",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Not on Movie Rolls? Sign up",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
